src/finufft_core.cpp

```diff
@@ -0,0 +1,1204 @@
+#include <finufft/fft.h>
+#include <finufft/finufft_core.h>
+#include <finufft/spreadinterp.h>
+#include <finufft/utils.h>
+
+#include "../contrib/legendre_rule_fast.h"
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+using namespace std;
```

Comment:
I don't particularly like doing this even in .cpp files because of the clash between cmath and compiler internal functions. I never know if I am calling std::log or log internal to the compiler.

Reply:
(although to be fair, only the occurrence in `finufft_core.cpp` was added by me :-)

Reply:
I've removed the occurrence in `finufft_core.cpp`; I'd rather not open this can or worms for the other files right now, especially `spreadinterp.cpp`.

Reply:
Sounds good :)